src/env.ts

```diff
@@ -0,0 +1,234 @@
+   */
+  PUBLIC_GATEWAY_IPFS: string;
+  /**
+   * Base URL for a public gateway which will provide access to all Arweave resources. Defaults to
+   * `https://arweave.net`.
+   */
+  PUBLIC_GATEWAY_ARWEAVE: string;
+}
+
+export function getEnvVars(): Env {
+  const schema = {
+    type: 'object',
+    required: [
+      'API_HOST',
+      'API_PORT',
+      'PGHOST',
+      'PGPORT',
+      'PGUSER',
+      'PGPASSWORD',
+      'PGDATABASE',
+      'BLOCKCHAIN_API_PGHOST',
+      'BLOCKCHAIN_API_PGPORT',
+      'BLOCKCHAIN_API_PGUSER',
+      'BLOCKCHAIN_API_PGPASSWORD',
+      'BLOCKCHAIN_API_PGDATABASE',
+      'STACKS_NODE_RPC_HOST',
+      'STACKS_NODE_RPC_PORT',
+      'PUBLIC_GATEWAY_IPFS',
+      'PUBLIC_GATEWAY_ARWEAVE',
+    ],
+    properties: {
+      API_HOST: {
+        type: 'string',
+      },
+      API_PORT: {
+        type: 'number',
+        default: 3000,
+        minimum: 0,
+        maximum: 65535,
+      },
+      PGHOST: {
+        type: 'string',
+      },
+      PGPORT: {
+        type: 'number',
+        default: 5432,
+        minimum: 0,
+        maximum: 65535,
+      },
+      PGUSER: {
+        type: 'string',
+      },
+      PGPASSWORD: {
+        type: 'string',
+      },
+      PGDATABASE: {
+        type: 'string',
+      },
+      PG_CONNECTION_POOL_MAX: {
+        type: 'number',
+        default: 10,
+      },
+      PG_IDLE_TIMEOUT: {
+        type: 'number',
+        default: 30,
+      },
+      PG_MAX_LIFETIME: {
+        type: 'number',
+        default: 60,
+      },
+      BLOCKCHAIN_API_PGHOST: {
+        type: 'string',
+      },
+      BLOCKCHAIN_API_PGPORT: {
+        type: 'number',
+        default: 5432,
+        minimum: 0,
+        maximum: 65535,
+      },
+      BLOCKCHAIN_API_PGUSER: {
+        type: 'string',
+      },
+      BLOCKCHAIN_API_PGPASSWORD: {
+        type: 'string',
+      },
+      BLOCKCHAIN_API_PGDATABASE: {
+        type: 'string',
+      },
+      BLOCKCHAIN_API_PG_CONNECTION_POOL_MAX: {
+        type: 'number',
+        default: 10,
+      },
+      BLOCKCHAIN_API_PG_IDLE_TIMEOUT: {
+        type: 'number',
+        default: 30,
+      },
+      BLOCKCHAIN_API_PG_MAX_LIFETIME: {
+        type: 'number',
+        default: 60,
+      },
+      STACKS_NODE_RPC_HOST: {
+        type: 'string',
+      },
+      STACKS_NODE_RPC_PORT: {
+        type: 'number',
+        minimum: 0,
+        maximum: 65535,
+      },
+      JOB_QUEUE_STRICT_MODE: {
+        type: 'boolean',
+        default: true,
+      },
+      JOB_QUEUE_MAX_RETRIES: {
+        type: 'number',
+        default: 5,
+      },
+      METADATA_FETCH_TIMEOUT_MS: {
+        type: 'number',
+        default: 10_000,
+      },
+      METADATA_MAX_PAYLOAD_BYTE_SIZE: {
+        type: 'number',
+        default: 1_000_000, // 1 MB
+      },
+      METADATA_MAX_NFT_CONTRACT_TOKEN_COUNT: {
+        type: 'number',
+        default: 50_000,
+      },
+      METADATA_IMAGE_CACHE_PROCESSOR: {
```

Comment:
Might be useful to specify an `example` here, `config/image-cache.js` right? I don't see that script referenced anywhere else, maybe would be useful to add to the docs?

Reply:
Done